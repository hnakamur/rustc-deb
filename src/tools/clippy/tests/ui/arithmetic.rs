#![warn(clippy::integer_arithmetic, clippy::float_arithmetic)]
#![allow(
    unused,
    clippy::shadow_reuse,
    clippy::shadow_unrelated,
    clippy::no_effect,
    clippy::unnecessary_operation
)]

#[rustfmt::skip]
fn main() {
    let i = 1i32;
    1 + i;
    i * 2;
    1 %
    i / 2; // no error, this is part of the expression in the preceding line
    i - 2 + 2 - i;
    -i;

    i & 1; // no wrapping
    i | 1;
    i ^ 1;
    i >> 1;
    i << 1;

    let f = 1.0f32;

    f * 2.0;

    1.0 + f;
    f * 2.0;
    f / 2.0;
    f - 2.0 * 4.2;
    -f;

    // No errors for the following items because they are constant expressions
    enum Foo {
        Bar = -2,
    }
    struct Baz([i32; 1 + 1]);
    union Qux {
        field: [i32; 1 + 1],
    }
    type Alias = [i32; 1 + 1];

    const FOO: i32 = -2;
    static BAR: i32 = -2;

    let _: [i32; 1 + 1] = [0, 0];

    let _: [i32; 1 + 1] = {
        let a: [i32; 1 + 1] = [0, 0];
        a
    };

    trait Trait {
        const ASSOC: i32 = 1 + 1;
    }

    impl Trait for Foo {
        const ASSOC: i32 = {
            let _: [i32; 1 + 1];
            fn foo() {}
            1 + 1
        };
    }
}
