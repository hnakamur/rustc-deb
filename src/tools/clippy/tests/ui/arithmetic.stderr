error: integer arithmetic detected
  --> $DIR/arithmetic.rs:13:5
   |
LL |     1 + i;
   |     ^^^^^
   |
   = note: `-D clippy::integer-arithmetic` implied by `-D warnings`

error: integer arithmetic detected
  --> $DIR/arithmetic.rs:14:5
   |
LL |     i * 2;
   |     ^^^^^

error: integer arithmetic detected
  --> $DIR/arithmetic.rs:15:5
   |
LL | /     1 %
LL | |     i / 2; // no error, this is part of the expression in the preceding line
   | |_________^

error: integer arithmetic detected
  --> $DIR/arithmetic.rs:17:5
   |
LL |     i - 2 + 2 - i;
   |     ^^^^^^^^^^^^^

error: integer arithmetic detected
  --> $DIR/arithmetic.rs:18:5
   |
LL |     -i;
   |     ^^

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:28:5
   |
LL |     f * 2.0;
   |     ^^^^^^^
   |
   = note: `-D clippy::float-arithmetic` implied by `-D warnings`

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:30:5
   |
LL |     1.0 + f;
   |     ^^^^^^^

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:31:5
   |
LL |     f * 2.0;
   |     ^^^^^^^

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:32:5
   |
LL |     f / 2.0;
   |     ^^^^^^^

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:33:5
   |
LL |     f - 2.0 * 4.2;
   |     ^^^^^^^^^^^^^

error: floating-point arithmetic detected
  --> $DIR/arithmetic.rs:34:5
   |
LL |     -f;
   |     ^^

error: aborting due to 11 previous errors

